src/screens/quick-stake-select-validator/quick-stake-select-validator.tsx

```diff
@@ -0,0 +1,141 @@
+import { withTheme, IThemeProps } from '../../core/theme/with-theme';
+import { connect } from 'react-redux';
+import { smartConnect } from '../../core/utils/smart-connect';
+import { IReduxState } from '../../redux/state';
+import { translate } from '../../core/i18n';
+import { SmartScreen } from '../smart-screen/smart-screen';
+import { ContextScreen } from '../../components/widgets/types';
+import { BottomCta } from '../../components/bottom-cta/bottom-cta';
+import { getSelectedAccount, getSelectedWallet } from '../../redux/wallets/selectors';
+import { getChainId } from '../../redux/preferences/selectors';
+import { getScreenDataKey } from '../../redux/ui/screens/data/reducer';
+import { buildDummyValidator } from '../../redux/wallets/actions';
+import {
+    navigateToEnterAmountStep,
+    QUICK_DELEGATE_ENTER_AMOUNT
+} from '../../redux/ui/screens/posActions/actions';
+import { IAccountState, ITokenState } from '../../redux/wallets/state';
+import { getBlockchain } from '../../core/blockchain/blockchain-factory';
+import { PrimaryCtaField } from '../../components/bottom-cta/primary-cta-field/primary-cta-field';
+import { valuePrimaryCtaField } from '../../core/utils/format-string';
+import { IValidator } from '../../core/blockchain/types/stats';
+
+interface IReduxProps {
+    validators: {
+        id: string;
+        name: string;
+    }[];
+    account: IAccountState;
+    token: ITokenState;
+    navigateToEnterAmountStep: typeof navigateToEnterAmountStep;
+}
+
+const mapStateToProps = (state: IReduxState) => {
+    const account = getSelectedAccount(state);
+    const chainId = getChainId(state, account.blockchain);
+
+    const screenKey = getScreenDataKey({
+        pubKey: getSelectedWallet(state)?.walletPublicKey,
+        blockchain: account?.blockchain,
+        chainId: String(chainId),
+        address: account?.address,
+        tab: undefined
+    });
+
+    return {
+        validators: state.ui.screens.inputData[screenKey]?.validators || [],
+        account,
+        token: account.tokens[chainId][getBlockchain(account.blockchain).config.coin]
+    };
+};
+
+const mapDispatchToProps = {
+    navigateToEnterAmountStep
+};
+
+const navigationOptions = () => ({ title: translate('App.labels.quickStake') });
+
+interface IState {
+    selectedValidators: IValidator[];
+}
+
+class QuickStakeSelectValidatorScreenComponent extends React.Component<
+    IReduxProps & IThemeProps<ReturnType<typeof stylesProvider>>,
+    IState
+> {
+    public static navigationOptions = navigationOptions;
+
+    constructor(props: IReduxProps & IThemeProps<ReturnType<typeof stylesProvider>>) {
+        super(props);
+        this.state = {
+            selectedValidators: []
+        };
+    }
+
+    public componentDidUpdate(prevProps: IReduxProps) {
+        if (this.props.validators && this.props.validators !== prevProps.validators) {
+            const selectedValidators = [];
+            for (const v of this.props.validators) {
```

Comment:
move this for inside `onPress` function from bottom cta component, this way you don't need local state anymore, less work less reders ;)